cloud/openstack/os_server_group.py

```diff
@@ -0,0 +1,182 @@
+          group. The list must contain at least one policy name. The current
+          valid policy names are anti-affinity, affinity, soft-anti-affinity
+          and soft-affinity.
+     required: false
+requirements:
+    - "python >= 2.6"
+    - "shade"
+'''
+
+EXAMPLES = '''
+# Create a server group with 'affinity' policy.
+- os_server_group:
+    state: present
+    auth:
+      auth_url: https://api.cloud.catalyst.net.nz:5000/v2.0
+      username: admin
+      password: admin
+      project_name: admin
+    name: my_server_group
+    policies:
+      - affinity
+
+# Delete 'my_server_group' server group.
+- os_server_group:
+    state: absent
+    auth:
+      auth_url: https://api.cloud.catalyst.net.nz:5000/v2.0
+      username: admin
+      password: admin
+      project_name: admin
+    name: my_server_group
+'''
+
+RETURN = '''
+id:
+    description: Unique UUID.
+    returned: success
+    type: string
+name:
+    description: The name of the server group.
+    returned: success
+    type: string
+policies:
+    description: A list of one or more policy names of the server group.
+    returned: success
+    type: list of strings
+members:
+    description: A list of members in the server group.
+    returned: success
+    type: list of strings
+metadata:
+    description: Metadata key and value pairs.
+    returned: success
+    type: dict
+project_id:
+    description: The project ID who owns the server group.
+    returned: success
+    type: string
+user_id:
+    description: The user ID who owns the server group.
+    returned: success
+    type: string
+'''
+
+
+def _system_state_change(state, server_group):
+    if state == 'present' and not server_group:
+        return True
+    if state == 'absent' and server_group:
+        return True
+
+    return False
+
+
+def main():
+    argument_spec = openstack_full_argument_spec(
+        name=dict(required=True),
+        policies=dict(required=False, type='list'),
+        state=dict(default='present', choices=['absent', 'present']),
+    )
+    module_kwargs = openstack_module_kwargs()
+    module = AnsibleModule(
+        argument_spec,
+        supports_check_mode=True,
+        **module_kwargs
+    )
+
+    if not HAS_SHADE:
+        module.fail_json(msg='shade is required for this module')
+
+    name = module.params['name']
+    policies = module.params['policies']
+    state = module.params['state']
+
+    try:
+        cloud = shade.openstack_cloud(**module.params)
+        server_group = cloud.get_server_group(name)
+
+        if module.check_mode:
+            module.exit_json(
+                changed=_system_state_change(state, server_group)
+            )
+
+        changed = False
+        if state == 'present':
+            if not server_group:
+                if not policies:
+                    module.fail_json(
+                        msg="Parameter 'policies' is required in Server Group "
+                            "Create"
```

Comment:
Thanks, @j2sol, I will update as you suggested if a new patchset is really needed.

BTW, what is next step after 'shipit'? I'm asking because I'm not familiar with the Ansible contribution process as I came from OpenStack community :-)